step-grid-agent/src/main/java/step/grid/agent/RegistrationTask.java

```diff
@@ -53,7 +53,8 @@ public void run() {
 	}
 	
 	protected void unregister() {
-		// TODO IMPLEMENT
+		logger.info("Switching agent to maintenance mode to prevent new incoming sessions....");
+		client.switchTokensToMaintenanceMode(agent.getTokens());

```

Comment:
Couldn't we use the service rest/grid/agent/3e0a4bb7-6ec4-44e8-b89b-c3b63f8bb582/interrupt to perform only one requests rather than 1 per token?

Reply:
@david-stephan I checked and unfortunately this service is a controller service from Step and therefore not part of the grid project